pkg/k8s/labels.go

```diff
@@ -209,6 +209,10 @@ const (
 	// ProxyGIDAnnotation can be used to override the GID config.
 	ProxyGIDAnnotation = ProxyConfigAnnotationsPrefix + "/proxy-gid"
 
+	// ProxyEnableShutdownEndpointAnnotation can be used to override the
+	// LINKERD2_PROXY_SHUTDOWN_ENDPOINT_ENABLED config.
+	ProxyEnableShutdownEndpointAnnotation = ProxyConfigAnnotationsPrefix + "/enable-shutdown-endpoint"
```

Comment:
As discussed, I think we probably ought to use something like
```suggestion
	ProxyEnableShutdownEndpointAnnotation = ProxyConfigAnnotationsPrefix + "/proxy-admin-shutdown"
```

with a value of 'enabled'.

1. `true` causes YAML weirdness, since the value must be explicitly quoted to avoid type errors in annotations. `enabled` is properly handled a string value without quotes.
2. Consistent proxy- prefixes sort together and help express scoping/grouping.